core/dbt/task/clean.py

```diff
@@ -1,54 +1,67 @@
 import os.path
 import os
 import shutil
+from typing import Any, Dict, List
 
 from dbt import deprecations
-from dbt.task.base import BaseTask, move_to_nearest_project_dir
+from dbt.config import Project
+from dbt.config.runtime import get_project_and_cli_vars_from_args
 from dbt.events.functions import fire_event
 from dbt.events.types import (
     CheckCleanPath,
     ConfirmCleanPath,
     ProtectedCleanPath,
     FinishedCleanPaths,
 )
-from dbt.config import UnsetProfileConfig
+from dbt.task.base import (
+    BaseTask,
+    get_nearest_project_dir,
+    move_to_nearest_project_dir,
+)
 
 
 class CleanTask(BaseTask):
-    # Note: CleanTask is the last task that uses UnsetProfileConfig,
-    # and can be deleted once CleanTask no longer requires it.
-    ConfigType = UnsetProfileConfig
-
-    def __is_project_path(self, path):
-        proj_path = os.path.abspath(".")
-        return not os.path.commonprefix([proj_path, os.path.abspath(path)]) == proj_path
-
-    def __is_protected_path(self, path):
-        """
-        This function identifies protected paths, so as not to clean them.
-        """
-        abs_path = os.path.abspath(path)
-        protected_paths = self.config.model_paths + self.config.test_paths + ["."]
-        protected_abs_paths = [os.path.abspath(p) for p in protected_paths]
-        return abs_path in set(protected_abs_paths) or self.__is_project_path(abs_path)
-
     def run(self):
         """
         This function takes all the paths in the target file
         and cleans the project paths that are not protected.
         """
         move_to_nearest_project_dir(self.args.project_dir)
         if (
-            "dbt_modules" in self.config.clean_targets
+            "dbt_modules" in self.project.clean_targets
             and self.config.packages_install_path not in self.config.clean_targets
         ):
             deprecations.warn("install-packages-path")
-        for path in self.config.clean_targets:
+        for path in self.project.clean_targets:
             fire_event(CheckCleanPath(path=path))
-            if not self.__is_protected_path(path):
+            if not is_protected_path(path, self.config.model_paths, self.config.test_paths):
                 shutil.rmtree(path, True)
                 fire_event(ConfirmCleanPath(path=path))
             else:
                 fire_event(ProtectedCleanPath(path=path))
 
         fire_event(FinishedCleanPaths())
+
+    @classmethod
+    def from_project(cls, project: Project, cli_vars: Dict[str, Any]) -> "CleanTask":
+        return cls(cli_vars, project)
+
+    @classmethod
+    def from_args(cls, args) -> "CleanTask":
```

Comment:
What is this used by?  I don't see it being called anywhere...

Reply:
Removed `from_args` from `CleanTask`.